openwisp_controller/config/base/config.py

```diff
@@ -163,15 +163,20 @@ def _get_templates_from_pk_set(cls, pk_set):
         return templates
 
     @classmethod
-    def clean_templates(cls, action, instance, pk_set, **kwargs):
+    def clean_templates(cls, action, instance, pk_set, raw_data={}, **kwargs):
```

Comment:
never used mutable structures as default arguments, because the default argument can change if other parts of the code modify that structure and this can generate nasty bugs.
Use `None`, then in the line which you need a dict you can instantiate a new dict if `None` is passed, eg:
```python
raw_data = raw_data or {}
```